homeassistant/components/climate/honeywell.py

```diff
@@ -210,6 +210,13 @@ def update(self):
             self._name = data['name']
             self._is_dhw = False
 
+        # The underlying library doesn't expose the thermostat's mode
+        # but we can pull it out of the big dictionary of information.
+        device = self.client.named_devices[self.name]
```

Comment:
I don't think we can assume that 'mode' will be available. Unfortunately I can't test as I don't have a DHW device.